web3/types.py

```diff
@@ -283,7 +283,7 @@ class GethSyncingSubscriptionResponse(SubscriptionResponse):
 
 class RPCResponse(TypedDict, total=False):
     error: Union[RPCError, str]
-    id: int
+    id: Union[int, str]
```

Comment:
I believe this should be `Optional[Union[int, str]]`. Sorry if my previous comment was misleading. We sometimes don't get an `id`, making this param `Optional`.